pandas/core/common.py

```diff
@@ -232,7 +233,9 @@ def asarray_tuplesafe(values: Iterable, dtype: NpDtype | None = None) -> ArrayLi
     elif isinstance(values, ABCIndex):
         return values._values
 
-    if isinstance(values, list) and dtype in [np.object_, object]:
+    if isinstance(values, list) and (
+        dtype in [np.object_, object] or any(is_list_like(val) for val in values)
```

Comment:
Any idea about performance impact here?

Reply:
Yeah in the worst case where this can't short circuit there's a perf hit

```
In [1]: values = list(range(1000))

In [2]: %timeit pd.core.common.asarray_tuplesafe(values)
279 µs ± 1.6 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [2]: %timeit pd.core.common.asarray_tuplesafe(values)
41.8 µs ± 623 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

Reply:
Is this a common use-case?

Would try-except be significantly faster?

Reply:
Good idea. With the try except I get closer to main performance (~44)